src/web/status.rs

```diff
@@ -0,0 +1,151 @@
+use super::cache::CachePolicy;
+use crate::{
+    db::Pool,
+    utils::spawn_blocking,
+    web::{error::AxumResult, match_version_axum},
+};
+use axum::{
+    extract::{Extension, Path},
+    http::header::ACCESS_CONTROL_ALLOW_ORIGIN,
+    response::IntoResponse,
+    Json,
+};
+
+pub(crate) async fn status_handler(
+    path: Path<(String, String)>,
+    pool: Extension<Pool>,
+) -> impl IntoResponse {
+    async fn inner(
```

Comment:
could you elaborate on the reasoning for this `inner` method ? Wouldn't this work directly in the handler too?

Reply:
I realised just an async block would actually work and added a comment, it's so that we apply the headers to both successful and failed responses.